src/Parser/PotreeBinParser.js

```diff
@@ -1,52 +1,118 @@
+        normalized: true,
+    },
+    CLASSIFICATION: {
+        numElements: 1,
+        arrayType: Uint8Array,
+        attributeName: 'classification',
+    },
+    // Note: at the time of writing, PotreeConverter will only generate normals in Oct16 format
+    // see PotreeConverter.cpp:121
+    // we keep all the historical value to still supports old conversion
+    NORMAL_SPHEREMAPPED: {
+        numElements: 2,
+        arrayType: Uint8Array,
+        attributeName: 'sphereMappedNormal',
+    },
+    // see https://web.archive.org/web/20150303053317/http://lgdv.cs.fau.de/get/1602
+    NORMAL_OCT16: {
+        numElements: 2,
+        arrayType: Uint8Array,
+        attributeName: 'oct16Normal',
+    },
+    NORMAL: {
+        numElements: 3,
+        arrayType: Float32Array,
+        attributeName: 'normal',
+    },
+};
+
 export default {
     /** @module PotreeBinParser */
     /** Parse .bin PotreeConverter format and convert to a THREE.BufferGeometry
      * @function parse
      * @param {ArrayBuffer} buffer - the bin buffer.
+     * @param {Object} pointAttributes - the point attributes information contained in layer.metadata coming from cloud.js
      * @return {Promise} - a promise that resolves with a THREE.BufferGeometry.
      *
      */
-    parse: function parse(buffer) {
+    parse: function parse(buffer, pointAttributes) {
         if (!buffer) {
             throw new Error('No array buffer provided.');
         }
 
         const view = new DataView(buffer);
         // Format: X1,Y1,Z1,R1,G1,B1,A1,[...],XN,YN,ZN,RN,GN,BN,AN
-        const numPoints = Math.floor(buffer.byteLength / 16);
-
-        const positions = new Float32Array(3 * numPoints);
-        const colors = new Uint8Array(4 * numPoints);
+        let pointByteSize = 0;
+        for (const potreeName of pointAttributes) {
+            const attr = POINT_ATTTRIBUTES[potreeName];
+            pointByteSize += attr.numElements * (attr.numByte || attr.arrayType.BYTES_PER_ELEMENT);
+        }
+        const numPoints = Math.floor(buffer.byteLength / pointByteSize);
 
-        const box = new THREE.Box3();
-        box.min.set(Infinity, Infinity, Infinity);
-        box.max.set(-Infinity, -Infinity, -Infinity);
-        const tmp = new THREE.Vector3();
+        const attrs = [];
+        // get the variable attributes
+        for (const potreeName of pointAttributes) {
+            const attr = POINT_ATTTRIBUTES[potreeName];
+            const numByte = attr.numByte || attr.arrayType.BYTES_PER_ELEMENT;
+            attrs.push({
+                potreeName,
+                numElements: attr.numElements,
+                attributeName: attr.attributeName,
+                normalized: attr.normalized,
+                array: new attr.arrayType(attr.numElements * numPoints),
+                numByte,
+                // Potree stores everything as int, and uses scale + offset
+                fnName: `getUint${numByte * 8}`,
+            });
+        }
```

Comment:
Apart from `array`, this could be done statically once and for all in the global scope

Reply:
I think I don't see what you mean, because I feel I can't. I'm constructing an array of attr that depends on pointAttributes. The point of this code is to copy the `POINT_ATTTRIBUTES` static array.